feat: Reintroduce LaxRedirectStrategy in HttpClient 5.3.x

otavioprado opened the pull request:
This commit adds a new implementation of the LaxRedirectStrategy, bringing back functionality similar to that found in HttpClient 4.x. The LaxRedirectStrategy automatically redirects all HEAD, GET, POST, and DELETE requests, relaxing the restrictions imposed by the HTTP specification on automatic redirection of POST methods.

### Key Features:
- **Supported Methods**: Redirects HEAD, GET, POST, and DELETE methods.
- **Location Resolution**: Automatically resolves relative redirect URIs against the request URI.
- **Compatibility**: Ensures compatibility with existing 4.x applications by providing a familiar API and redirect behavior.
- **Normalization**: Normalizes and converts the host part of the redirect URI to lowercase, maintaining consistent behavior.

### Benefits:
- Simplifies migration from HttpClient 4.x to 5.x by providing a drop-in replacement for LaxRedirectStrategy.
- Supports applications that rely on flexible redirect behavior for non-idempotent HTTP methods.

### Usage:
```java
CloseableHttpClient httpClient = HttpClients.custom()
    .setRedirectStrategy(LaxRedirectStrategy.INSTANCE)
    .build();

ok2c reviewed (changes requested):
@otavioprado Few minor things
1. Generally we do not add new features (and this is technically a new feature) to stable branches. Would you mind changing the target branch to `master` (5.4)?
2. Please change `@since` tag to `5.4` 
3. Test methods should have package visibility for consistency with existing test cases

otavioprado commented:
> @otavioprado Few minor things
> 
> 1. Generally we do not add new features (and this is technically a new feature) to stable branches. Would you mind changing the target branch to `master` (5.4)?
> 2. Please change `@since` tag to `5.4`
> 3. Test methods should have package visibility for consistency with existing test cases

Hi @ok2c,

I've addressed your comments and reopened the pull request with the necessary changes. The new PR is targeting the master branch as requested, and you can find it here: [#565](https://github.com/apache/httpcomponents-client/pull/565).

Thank you for your feedback, and please let me know if there are any additional adjustments needed!
